### DISCUSSION TITLE
Assessment of the difficulty in porting CPU architecture for highfive

### DISCUSSION THREAD

**Discussion by wangyuliu** [BlueBrain/HighFive#853] [General]
Hello everyone! I am working on implementing a tool to assess the complexity of CPU architecture porting. It primarily focuses on RISC-V architecture porting. In fact, the tool may have an average estimate of various architecture porting efforts.My focus is on the overall workload and difficulty of transplantation in the past and future,even if a project has already been ported.As part of my dataset, I have collected the **HighFive** project. **I would like to gather community opinions to support my assessment. I appreciate your help and response!** Based on scanning tools, the porting complexity is determined to be simple, with a small amount of code related to the CPU architecture in the project.  Is this assessment accurate?Do you often have any opinions on personnel allocation and consumption time？ I look forward to your help and response.

**Comment by 1uc** [comment]
There's an open ticket about 32-bit not working, see:
https://github.com/BlueBrain/HighFive/issues/443

Then there's build system related pain. Tickets claim there's issues with very specific setups involving Windows, bash and ninja (2 years old and likely resolved by now), and a recent report that it doesn't work on FreeBSD. The project uses a lot of C++ templates. Hence, you'll need a C++ compiler that's not buggy. Finally, there's the issue that HighFive is only a wrapper for HDF5, it does nothing other than call HDF5. Hence, for HighFive to be portable you'd need to port HDF5.

Sorry, I don't know much about porting HighFive. Hopefully, others can help you.
